lib/ansible/modules/database/postgresql/postgresql_sequence.py

```diff
@@ -0,0 +1,744 @@
+    default: present
+    choices: [ absent, present ]
+    type: str
+  data_type:
+    description:
+    - Specifies the data type of the sequence. Valid types are bigint, integer, and smallint. bigint is the default.
+      The data type determines the default minimum and maximum values of the sequence.
+    - Supported from PostgreSQL 10.
+    required: false
+    choices: [ bigint, integer, smallint ]
+    type: str
+  increment:
+    description:
+    - Increment specifies which value is added to the current sequence value to create a new value.
+    - A positive value will make an ascending sequence, a negative one a descending sequence. The default value is 1.
+    required: false
+    type: int
+  minvalue:
+    description:
+    - Minvalue determines the minimum value a sequence can generate. The default for an ascending sequence is 1.
+      The default for a descending sequence is the minimum value of the data type.
+    required: false
+    type: int
+  maxvalue:
+    description:
+    - Maxvalue determines the maximum value for the sequence. The default for an ascending sequence is the maximum
+      value of the data type. The default for a descending sequence is -1.
+    required: false
+    type: int
+  start:
+    description:
+    - Start allows the sequence to begin anywhere. The default starting value is
+      I(minvalue) for ascending sequences and I(maxvalue) for descending ones.
+    required: false
+    type: int
+  cache:
+    description:
+    - Cache specifies how many sequence numbers are to be preallocated and
+      stored in memory for faster access. The minimum value is 1 (only one
+      value can be generated at a time, i.e., no cache), and this is also
+      the default.
+    required: false
+    type: int
+  cycle:
+    description:
+    - The cycle option allows the sequence to wrap around when the I(maxvalue) or I(minvalue) has been reached by an
+      ascending or descending sequence respectively. If the limit is reached, the next number generated will be the
+      minvalue or maxvalue, respectively.
+    - If false (NO CYCLE) is specified, any calls to nextval after the sequence has reached its maximum value will
+      return an error. False (NO CYCLE) is the default.
+    required: false
+    type: bool
+  cascade:
+    description:
+    - Automatically drop objects that depend on the sequence, and in turn all
+      objects that depend on those objects.
+    - Only used when I(state=absent).
+    required: false
+    type: bool
+  restrict:
+    description:
+    - Refuse to drop the sequence if any objects depend on it. This is the default.
+    - Only used when I(state=absent).
+    required: false
+    type: bool
+  rename_to:
+    description:
+    - The new name for the I(sequence).
+    - Works only for existing sequences.
+    required: false
+    type: str
+  owner:
+    description:
+    - Set the owner for the I(sequence).
+    - Works only for existing sequences.
```

Comment:
IMO, we should implement it for new sequences too

Reply:
ATM I see no option for this in the [documentation](https://www.postgresql.org/docs/current/sql-createsequence.html). Maybe I'm wrong but there is only the option `OWNED BY` which can be used to set a table as a owner (as a link to use a sequence as an indexer).

Reply:
It would be blown up the code, but one option could be: first create the sequence and then change the owner directly after.

Reply:
if owner passed you could run ALTER SEQUENS blabla OWNER TO blabla and add this to executed_queries.
See line 440 of postgresql_table module

Reply:
looks quite simple to me

Reply:
like this (in postgresql_table):
```
440     def set_owner(self, username):
441         query = "ALTER TABLE %s OWNER TO %s" % (pg_quote_identifier(self.name, 'table'),
442                                                 pg_quote_identifier(username, 'role'))
443         self.executed_queries.append(query)
444         return self.__exec_sql(query, ddl=True)
```
otherwise we must run this playbook twice to set another owner than login_role (it might happen very often) for a new sequence.
better to add this now

Reply:
Maybe with option `session_role` it will be possible to set this before or not?
But I like your approach and will add this.

Reply:
Anyway, IMO, better to do it by owner= during creation.
If there are any difficulties, I could  implement it later, but I’d like to see it added now ;)

Reply:
Anyway, IMO, better to do it by owner= during creation.
If there are any difficulties, I could  implement it later, but I’d like to see it added now ;)

Reply:
I changed that, so that owner can be set during creation and added one test case for that.